### PR TITLE
Added wishlist and cart with context.

### PR DESCRIPTION
**Pull request by dhruvin25799**
## Added Wishlist and Cart Pages with context.
Live : https://chimerical-liger-9b95c2.netlify.app/

## Functionality to check : 

- [ ] Cart and wishlist state management
- [ ] Add to cart and add to wishlist vice versa

## Files to check : 

- [ ] src/components/CartList/CartList.jsx
- [ ] src/components/CartTotals/CartTotals.jsx
- [ ] src/context/cart-context.js
- [ ] src/context/wishlist-context.js
- [ ] src/pages/Cart/Cart.jsx
- [ ] src/pages/Wishlist/Wishlist.jsx
- [ ] src/reducers/CartReducer.js
- [ ] src/reducers/wishlistReducer.js

## Files to ignore : 

- [ ] CSS files as they've been previously reviewed.
- [ ] Login/Signup Page, it is still WIP will complete with Auth after the lecture.

### CODE REVIEW

File: src/components/CartTotals/CartTotals.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            <hr />
+            <div className={styles["checkout-details"]}>
+              <div className={styles["checkout-detail-item"]}>
+                <p className={styles["detail-header"]}>Subtotal:</p>
+                <p>
+                  <RupeeSign/>
+                  {+cart.totalAmount}
+                </p>
+              </div>
+              {/* <div className={styles["checkout-detail-item"]}>
```
Comment (Ananyamadhu08) [inline src/components/CartTotals/CartTotals.jsx:23]: can delete commented code

File: src/components/CartTotals/CartTotals.module.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
+.cart-checkout-container{
+    flex: 1 1 30%;
+    padding: 1rem;
+}
+
+.cart-checkout{
+    padding: 1rem;
+    background-color: white;
```
Comment (Ananyamadhu08) [inline src/components/CartTotals/CartTotals.module.css:8]: use variable for color to maintain consistency.

File: src/components/ProductList/ProductList.jsx [hunk 3]
```diff
@@ -4,7 +4,23 @@ import { getAllProducts } from "../../helpers/getAllProducts";
...
 import { ProductCard } from "../Card/ProductCard";
+import { useCart } from "../../context/cart-context";
+import { useWishlist } from "../../context/wishlist-context";
 export const ProductList = (props) => {
+  const { cart, cartDispatch } = useCart();
+  const { wishlist, wishlistDispatch } = useWishlist();
+  const onAddToCart = (product) => {
+    cartDispatch({ type: "ADD", payload: product });
+  };
+  const onAddToWishlist = (product) => {
```
Comment (Ananyamadhu08) [inline src/components/ProductList/ProductList.jsx:15]: can remove the parentheses around the argument and the curly braces incase of one line arrow functions.

File: src/pages/Cart/Cart.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import styles from "./Cart.module.css";
+import { CartTotals } from "../../components/CartTotals/CartTotals";
+import { CartList } from "../../components/CartList/CartList";
+
+export const Cart = (props) => {
```
Comment (Ananyamadhu08) [inline src/pages/Cart/Cart.jsx:6]: destructing can be used

File: src/pages/Wishlist/Wishlist.module.css [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+}
+.banner-products.wishlist{
+    gap: 0.5rem;
+}
+
+.wishlist-main{
+    padding: 2rem;
+    display: flex;
+    flex-flow: column;
+    background-color: #F7F7F7;
```
Comment (Ananyamadhu08) [inline src/pages/Wishlist/Wishlist.module.css:47]: try using variables for colors throughout your application to maintain consistency.

File: src/components/CartTotals/CartTotals.jsx (hunk 1 above)
Comment (Ananyamadhu08) [inline src/components/CartTotals/CartTotals.jsx:23]: can delete the commented code here

**Review by Ananyamadhu08 — APPROVED** [review]
delete the commented code, and use consistent variables for color. Your UI is very good and your code is clean, you can go ahead and merge

File: src/pages/Signup/Signup.module.css [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+.login-box .login-input{
+    display: flex;
+    flex-flow: column;
+    gap: 0.5rem;
+}
+
+.login-input .input-control{
+    display: flex;
+    flex-flow: column;
+    gap: 0.3rem;
```
Comment (deekshithmd) [inline src/pages/Signup/Signup.module.css:81]: Suggestion : Better to use px for these small values, orelse when screen size decreased gap will become invisible

**Review by deekshithmd — APPROVED** [review]

**Pull request merged**